lib/instrumentation/index.js

```diff
@@ -414,24 +414,26 @@ Instrumentation.prototype.addEndedSpan = function (span) {
     // if I have ended and I have something buffered, send that buffered thing
     if (span.getBufferedSpan()) {
       this._encodeAndSendSpan(span.getBufferedSpan())
+      span.setBufferedSpan(null)
     }
 
-    if (!span.isCompressionEligible() || span.getParentSpan().ended) {
-      const buffered = span.getBufferedSpan()
+    const parentSpan = span.getParentSpan()
+    if ((parentSpan && parentSpan.ended) || !span.isCompressionEligible()) {
+      const buffered = parentSpan && parentSpan.getBufferedSpan()
       if (buffered) {
         this._encodeAndSendSpan(buffered)
-        span.setBufferedSpan(null)
+        parentSpan.setBufferedSpan(null)
```

Comment:
REVIEW NOTE: This corrects the logic to make the spec pseudo code: this block is meant to be able looking at the possibly buffered span on the *parent*.   This fixes the case where a buffered span would not get sent if it was followed by an *incompressible* sibling span.